src/main/java/org/apache/camel/idea/util/IdeaUtils.java

```diff
@@ -316,8 +318,13 @@ public static boolean isCamelExpressionOrLanguage(PsiClass clazz) {
     private static boolean isFromJavaMethod(PsiMethodCallExpression call, String... methods) {
         PsiMethod method = call.resolveMethod();
         if (method != null) {
-            String name = method.getName();
-            return Arrays.stream(methods).anyMatch(name::equals);
+            PsiClass containingClass = method.getContainingClass();
+            if (containingClass != null) {
+                String className = containingClass.getQualifiedName();
+                String name = method.getName();
+
+                return Arrays.stream(methods).anyMatch(name::equals) && "org.apache.camel.builder.RouteBuilder".equals(className);
```

Comment:
I didn't thought about that, now it is fixed. How about the XML route? Checking if <from> is a child of <route> is not enough, <route> must be a child of <camelContext>. What do you think?